### DISCUSSION TITLE
New connection forces window to change to different display

### DISCUSSION THREAD

**Discussion by cody151** [rustdesk/rustdesk#8825] [General]
### Bug Description

On previous version when I moved the rustdesk window to screen 2 it would stay there even when opening a new connection but the recent update has no broken something, when the window is moved to screen 2 and then a new connection is opened it forces the window to move to screen 1 for no reason. This never happened on older versions (maybe 1.2.3) - new update broke something.

### How to Reproduce

Multi-display set up, open a remote connection and move the rustdesk window to one screen (non-primary screen), then open a new remote connection. Window will force to the primary screen.

### Expected Behavior

Opening new connection should keep the window position (on non-primary screen) intact.

### Operating system(s) on local side and remote side

Windows 10 -> Windows 10

### RustDesk Version(s) on local side and remote side

1.2.7 -> 1.2.7

### Screenshots

Video or picture cannot describe issue as I do not want to expose sensitive RustDesk IDs

### Additional Context

Just open a remote connection and move the rustdesk connection window to a non-primary screen, then open a new connection and the rustdesk connection window will forcefully move to the primary screen - This never happened on previous version (maybe 1.2.3)

**Comment by rustdesk** [comment]
Apologize, without screenshot, I can not get what you are saying.

**Reply by cody151** [reply]
Sorry I do not feel comfortable to record the screen because it displays sensitive RustDesk ID of my remote computers

**Reply by cody151** [reply]
I have recorded the example, how can I send it to you privately

**Reply by dignow** [reply]
Is this the same problem I describe below?

**Reply by cody151** [reply]
> Is this the same problem I describe below?

Sorry it is confusing for me, I agree with @rustdesk it is very difficult to understand without screen recording

**Reply by cody151** [reply]
Please see the diagram I posted in another comment

**Reply by cody151** [reply]
@rustdesk it may seem like a simple issue but it gets very irritating after a while as it blocks other programs and disrupts view of primary screen

**Reply by rustdesk** [reply]
We won't do anything before seeing the video. We have no time to guess.

**Reply by cody151** [reply]
But you are asking me to expose sensitive rustdesk ID, this should be completely not recommended with regards to security, I have spent time to create the diagram below which you can see.

**Comment by dignow** [comment]
I recently modified the position strategy.

If there are two connections in a window, the window position will move to the position of the later connection, because the later connection is currently used. Eg.

The positions stored: `A` -> 30, `B` -> 13

If `A` is connected, the window position is 30.
Then the window can be moved to 56.
But when `B` is connected in new tab (not new window), the window is auto moved to 13.

**Comment by cody151** [comment]
@rustdesk please see the diagram attached 
![24gwereh34greh35h35rg3w4rg](https://github.com/user-attachments/assets/fd68f76b-b79b-4431-b94c-822907e5db8b)

**Reply by cody151** [reply]
Just to reiterate Version 1.2.3 did not have this issue!

**Reply by cody151** [reply]
@rustdesk check diagram I spent time to create

**Comment by fufesou** [comment]
https://github.com/rustdesk/rustdesk/pull/8834
